### ISSUE TITLE
bndtools.m2e: Different contracts between IDE and Maven

### ISSUE THREAD

**Issue by maggu2810** [bndtools/bnd#3378]
Version of plugins in the Eclipse IDE: 4.2.0.REL-20190305-163318-g7028683
Version of Maven plugins: 4.2.0

In a multi module project (cannot publish the sources) one project is using JPA.

If the project is build in the IDE the generated manifest in the JAR file contains a requirement for the contract JavaJPA version 2.2.

```
osgi.contract;osgi.contract=JavaJPA;filter:="(&(osgi.contract=JavaJPA)(version=2.2.0))"
```

If the project is build by Maven on command line the generated manifest in the JAR file contains a requirement for the contract JavaJPA version 2.1.

```
osgi.contract;osgi.contract=JavaJPA;filter:="(&(osgi.contract=JavaJPA)(version=2.1.0))"
```

The Eclipse IDE shows the following Maven dependencies for the project:

```
.../.m2/repository/org/osgi/osgi.core/6.0.0/osgi.core-6.0.0.jar
.../.m2/repository/org/osgi/osgi.cmpn/7.0.0/osgi.cmpn-7.0.0.jar
.../.m2/repository/org/osgi/osgi.annotation/7.0.0/osgi.annotation-7.0.0.jar
.../.m2/repository/org/apache/felix/org.apache.felix.http.servlet-api/1.1.2/org.apache.felix.http.servlet-api-1.1.2.jar
.../.m2/repository/org/apache/tomcat/tomcat-servlet-api/8.0.9/tomcat-servlet-api-8.0.9.jar
.../.m2/repository/org/apache/aries/spec/org.apache.aries.javax.jax.rs-api/1.0.0/org.apache.aries.javax.jax.rs-api-1.0.0.jar
.../.m2/repository/org/apache/geronimo/specs/geronimo-osgi-locator/1.0/geronimo-osgi-locator-1.0.jar
.../.m2/repository/org/apache/aries/jpa/javax/persistence/javax.persistence_2.1/2.7.2/javax.persistence_2.1-2.7.2.jar
.../.m2/repository/org/apache/felix/org.apache.felix.gogo.runtime/1.1.2/org.apache.felix.gogo.runtime-1.1.2.jar
.../.m2/repository/org/apache/servicemix/specs/org.apache.servicemix.specs.json-api-1.1/2.9.0/org.apache.servicemix.specs.json-api-1.1-2.9.0.jar
.../.m2/repository/org/apache/johnzon/johnzon-mapper/1.1.12/johnzon-mapper-1.1.12.jar
.../.m2/repository/org/apache/johnzon/johnzon-core/1.1.12/johnzon-core-1.1.12.jar
.../.m2/repository/org/slf4j/slf4j-api/1.7.21/slf4j-api-1.7.21.jar
.../.m2/repository/org/eclipse/jdt/org.eclipse.jdt.annotation/2.2.100/org.eclipse.jdt.annotation-2.2.100.jar
.../.m2/repository/com/github/spotbugs/spotbugs-annotations/3.1.12/spotbugs-annotations-3.1.12.jar
.../.m2/repository/com/google/code/findbugs/jsr305/3.0.2/jsr305-3.0.2.jar
```

The Maven dependency tree looks like:

```
[INFO] |  +- org.osgi:osgi.core:jar:6.0.0:provided
[INFO] |  +- org.osgi:osgi.cmpn:jar:7.0.0:provided
[INFO] |  +- org.osgi:osgi.annotation:jar:7.0.0:provided
[INFO] |  +- org.apache.felix:org.apache.felix.http.servlet-api:jar:1.1.2:provided
[INFO] |  |  \- org.apache.tomcat:tomcat-servlet-api:jar:8.0.9:provided
[INFO] |  +- org.apache.aries.spec:org.apache.aries.javax.jax.rs-api:jar:1.0.0:provided
[INFO] |  |  \- org.apache.geronimo.specs:geronimo-osgi-locator:jar:1.0:provided
[INFO] |  +- org.apache.aries.jpa.javax.persistence:javax.persistence_2.1:jar:2.7.2:provided
[INFO] |  +- org.apache.felix:org.apache.felix.gogo.runtime:jar:1.1.2:provided
[INFO] |  +- org.apache.servicemix.specs:org.apache.servicemix.specs.json-api-1.1:jar:2.9.0:provided
[INFO] |  +- org.apache.johnzon:johnzon-mapper:jar:1.1.12:provided
[INFO] |  |  \- org.apache.johnzon:johnzon-core:jar:1.1.12:provided
[INFO] |  +- org.slf4j:slf4j-api:jar:1.7.21:provided
[INFO] |  +- org.eclipse.jdt:org.eclipse.jdt.annotation:jar:2.2.100:provided
[INFO] |  \- com.github.spotbugs:spotbugs-annotations:jar:3.1.12:provided
[INFO] |     \- com.google.code.findbugs:jsr305:jar:3.0.2:provided
```

I don't see any usage of JPA 2.2 at all...

**Comment by maggu2810** [comment]
If I disable the JavaJPA contract, I see that the packages `javax.persistence` and `javax.persistence.criteria` are imported in version 1.3 (not 1.2 as expected).

**Comment by bjhargrave** [comment]
Can you please provide a small github repo which demonstrates the problem? Then we can investigate. Thanks.

**Comment by maggu2810** [comment]
That's strange.
I copied the whole project to create a stripped down version.
I imported the copy in another Eclipse IDE instance and removed all unrelated stuff.
I did not see the error anymore.

After that I imported the original project into the other Eclipse IDE instance -- the problem does not exist anymore.

I restarted the previous Eclipse IDE and don't see the problem anymore.

So, it is working now and I assume we cannot reproduce it for the moment.
But I still don't understand why JavaJPA 2.2 and java.persistence 1.3 has been used inside my project in the IDE at a state now dependent bundle provides it.
